lib/claim_letters/claim_letter_downloader.rb

```diff
@@ -66,12 +61,17 @@ def file_number
     end
 
     def filter_letters(document)
-      return nil unless allowed_doctypes.include?(document[:doc_type])
+      return nil unless @allowed_doctypes.include?(document[:doc_type])
 
-      document.marshal_dump
+      if document.is_a?(OpenStruct)
```

Comment:
What caused this change? Did you run into an error with the previous implementation?

Reply:
I added a check for `filter_letters(letter_details).nil?` in the `get_letter` method, and needed `filter_letters` to start accepting non-`OpenStruct` objects for it to work. @jerekshoe

Reply:
Adding that check resolves the open question in the ticket for this: `Should we be worried about filtering out based on docTypes when requesting documents by their ID?`

Reply:
This makes me wonder if we should move the marshal_dump stuff out of this method. I would at least add a comment about why we need to check

Reply:
@jerekshoe Ok, I'll refactor and/or comment on it. Thanks.